test/src/test/java/org/corfudb/integration/CorfuReplicationUpgradeIT.java

```diff
@@ -937,4 +938,119 @@ private void verifyVersion(CorfuStore corfuStore, String expectedVersion, boolea
         Assert.assertEquals(isUpgraded, actualUpgradedFlag);
         log.info("Verified version");
     }
+
+    @Test
+    public void testLocalClusterRollingUpgrade() throws Exception {
```

Comment:
I don't understand what this test is validating, as all 3 nodes share the same VERSION_STRING, how are we exercising rolling upgrade?... Also, this seems a  copy/paste of another test, can't we just modify that code a bit to accept if rollingUpgrade or not, cause at the end this is just testing replication E2E when one node is upgraded and not all, is it?

Reply:
I was just trying to add some code coverage tests but did not get to complete it yet. The actual code to exercise the api isn't here yet. Let me see if I can add something to perhaps unit test the calls instead.

Reply:
Fixed the test case to validate the new code added.